hack/gha-reversemap.sh

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Purpose: Harmonise '$GITHUB_WORKFLOWS_PATH' and the reversemap '$REVERSEMAP_FILE'
+
+# Usage: see help function
+# Working directory must be the project root
+
+# Define the required version of yq
+GITHUB_TOKEN=
+GITHUB_WORKFLOWS_PATH="./.github/workflows"
+REVERSEMAP_FILE=".gha-reversemap.yml"
+YQ_REQUIRED_VERSION="v4.45.1"
+GIT_COMMITSHA_LENGTH=40
+
+ERR_YQ_DOWNLOAD_FAILED=50
+ERR_YQ_NOT_INSTALLED=60
+ERR_GITHUB_TOKEN_INVALID=70
+ERR_ARCH_UNSUPPORTED=80
+
+help() {
+    cat <<EOF
+Harmonise '$GITHUB_WORKFLOWS_PATH' and the reversemap '$REVERSEMAP_FILE'
+
+Usage:
+    $0  <operation> [ARGUMENTS]
+
+Example:
+    # Update actions/checkout to its latest release
+    $0  update actions/checkout
+
+Operations:
+    apply-reversemap        [...WORKFLOW_FILE]          -   Apply the commit sha of every actions listed in the reversemap 
```

Comment:
(A) The usual way to write this would be `apply-reversemap [WORKFLOW_FILE...]` . I.e., put the ellipsis after the argument name.

(B) s/every actions/every action/

(C) the description says "all github workflows" but that's just one case. You could rewrite the description to something like the following.

> Update the given workflow files with the information in the reversemap file; if no workflow files are given then all are updated.

or

> Update the given workflow files, or all if none are given, with the information in the reversemap file.

Analogous comments apply to update-reversemap.